agent/command/git.go

```diff
@@ -78,6 +78,16 @@ type gitFetchProject struct {
 
 	RecurseSubmodules bool `mapstructure:"recurse_submodules"`
 
+	// CloneAllBranches specifies whether all branches should be cloned with the '--single-branch' flag.
```

Comment:
For monitoring, I want to add a fallback to the existing way (re-add it since I deleted it) and log to splunk that the new way failed- 

I've done some testing, this should not change any user-facing workflow and should only be a performance improvement but I agree that it would be nice to announce in the comms channel so there is more awareness just in case!

Reply:
A small heads up announcement SGTM!

It might be difficult to log to Splunk since this is in the agent, and only the app server can easily log to Splunk (you could do it, but it would have to be a critical or higher log).

Maybe adding a Honeycomb attribute to indicate when it's falling back to full clone would help? It's not perfect and Honeycomb wouldn't always show every full clone fallback due to sampling, but you could still see if it happens occasionally.

Reply:
I think building a tiny route to hit the app servers and log also wouldn't be the end of the world! Honeycomb sampling might not be preferable if you're trying to find edge cases.